rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/ResourceFormatterService.java

```diff
@@ -1,70 +1,139 @@
 package com.regnosys.rosetta.formatting2;
 
 import java.util.Collection;
+import java.util.function.BiConsumer;
 
 import org.eclipse.emf.ecore.resource.Resource;
 import org.eclipse.xtext.preferences.ITypedPreferenceValues;
 import org.eclipse.xtext.resource.XtextResource;
 
 public interface ResourceFormatterService {
-	
+
 	/**
-	 * Formats each {@link XtextResource} in the provided collection in-memory.
+	 * Formats each {@link XtextResource} in the provided collection in-memory and
+	 * passes the formatted content to a handler for further processing.
 	 * <p>
-	 * This method iterates over the given collection of resources and applies formatting 
-	 * directly to each resource. Formatting may include indentation, spacing adjustments, 
-	 * and other stylistic improvements to ensure consistency and readability of the resources.
+	 * This method iterates over the given collection of resources and applies
+	 * formatting directly to each resource. Formatting may include indentation,
+	 * spacing adjustments, and other stylistic improvements to ensure consistency
+	 * and readability of the resources.
 	 * </p>
 	 * 
-	 * @param resources a collection of {@link XtextResource} objects to be formatted
+	 * <p>
+	 * The handler, represented as a {@link java.util.function.BiConsumer}, is
+	 * called with two arguments: the {@link Resource} being formatted, and the
+	 * resulting formatted text as a {@link String}. This allows the caller to
+	 * specify actions such as saving the formatted content, logging it, or
+	 * collecting it for assertions in a test.
+	 * </p>
+	 * 
+	 * @param resources a collection of {@link XtextResource} objects to be
+	 *                  formatted
+	 * @param handler   a {@link java.util.function.BiConsumer} that processes each
+	 *                  formatted resource. The first argument is the
+	 *                  {@link Resource}, and the second is the formatted content as
+	 *                  a {@link String}.
 	 */
-	default void formatCollection(Collection<Resource> resources) {
-		formatCollection(resources, null);
+	default void formatCollection(Collection<Resource> resources, BiConsumer<Resource, String> handler) {
```

Comment:
Maybe it's mostly a style preference, but for readability I would prefer having a custom functional interface. E.g., `IFormattedResourceAcceptor`, which exposes a single `accept` method. That way, you could also document the inputs of the `accept` method separately. And it gives you the opportunity to define your own functional interface. ;)